src/shogun/mathematics/linalg/GPUMemoryBase.h

```diff
@@ -0,0 +1,21 @@
+#ifndef GPUMemory_Base_H__
+#define GPUMemory_Base_H__
+
+namespace shogun
+{
+
+template <typename T>
+struct GPUMemoryBase
+{
+	GPUMemoryBase()
+	{
+	}
+
+	virtual GPUMemoryBase<T>* clone_vector(const SGVector<T>& vector) const = 0;
+
+	virtual void from_gpu(T* data) const = 0;
```

Comment:
There's no `to_cpu` or `to_gpu` methods in `GPUMemoryBase` or `GPUMemoryViennaCL`.
Also, as you mentioned, if we do all the transfer in `LinalgBakcendViennaCL`,  we don't need `from_gpu` either. I'll remove it.